core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -301,6 +303,7 @@ public PlanOptimizers(
                 costCalculatorWithEstimatedExchanges,
                 costComparator,
                 taskCountEstimator,
+                createDistinctAggregationStrategyChooser(taskCountEstimator),
```

Comment:
Create directly this in the rules that need it -- just pass the TaskCountEstimator object. PlanOptimizers should not have to be aware about this implementation detail.